core/src/com/unciv/ui/utils/ExtensionFunctions.kt

```diff
@@ -360,15 +360,25 @@ fun (() -> Unit).wrapCrashHandlingUnit(
     return { wrappedReturning() ?: Unit }
 }
 
-/** For filters containing '{', apply the [predicate] to each part inside "{}" and aggregate using [operation];
- *  otherwise return `null` for Elvis chaining of the individual filter. */
-fun <T> String.filterCompositeLogic(predicate: (String) -> T?, operation: (T, T) -> T): T? {
-    val elements: List<T> = removePrefix("{").removeSuffix("}").split("} {")
+private val compositeLogicCheckRegex = Regex("""^\s*(?:\{[^{}]+}\s*)+$""")
+private val compositeLogicSplitRegex = Regex("""\{([^{}]+)}""")
```

Comment:
You'll need a \ before the final { - AS lies to you and says it's unneeded, but on Android it is needed. Same thing on the line above,

Reply:
I remember the comments elsewhere, yes. Sure that's still relevant, as we have bumped minimum versions since those days? Didn't the broken regex only appear on Java6-based Dalvik engines?  ... Still, you're right, stay defensive. After all, a similar fuzzy motivation keeps me from using `RegexOption.COMMENTS`.